These changes were added in 
<https://github.com/WebKit/webkit/commit/c7d6af038dcbfac95d0af4f43fbb945d73f6b1bb>

they don't compile, as they use EGL unprotected, but even when some of them were protected by USE(EGL) there were
 errors when trying to play videos in the MiniBrowser. So I added protection to all the addition bits in
 the commit that were not initially protected. 


diff --git Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.cpp Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.cpp
index ad632205..29a9eace 100644
--- Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.cpp
+++ Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.cpp
@@ -23,7 +23,7 @@
 #if USE(GSTREAMER)
 #include <gst/gst.h>
 
-#if USE(GSTREAMER_GL)
+#if USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 #include <gst/gl/egl/gsteglimage.h>
 #endif
 
@@ -508,6 +508,7 @@ template<> void derefGPtr<GstGLContext>(GstGLContext* ptr)
         gst_object_unref(GST_OBJECT(ptr));
 }
 
+#if USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 template <> GRefPtr<GstEGLImage> adoptGRef(GstEGLImage* ptr)
 {
     return GRefPtr<GstEGLImage>(ptr, GRefPtrAdopt);
@@ -525,6 +526,7 @@ template <> void derefGPtr<GstEGLImage>(GstEGLImage* ptr)
     if (ptr)
         gst_egl_image_unref(ptr);
 }
+#endif //USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 
 #endif // USE(GSTREAMER_GL)
 
--- Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.h.orig	2023-04-13 02:50:41.000000000 -0400
+++ Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.h	2023-05-16 14:43:42.000000000 -0400
@@ -31,7 +31,9 @@
 #if USE(GSTREAMER_GL)
 typedef struct _GstGLDisplay GstGLDisplay;
 typedef struct _GstGLContext GstGLContext;
+#if USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 typedef struct _GstEGLImage GstEGLImage;
+#endif  // USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 typedef struct _GstGLColorConvert GstGLColorConvert;
 #endif
 
@@ -48,9 +50,11 @@
 
 namespace WTF {
 
+#if USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 template<> GRefPtr<GstPlugin> adoptGRef(GstPlugin* ptr);
 template<> GstPlugin* refGPtr<GstPlugin>(GstPlugin* ptr);
 template<> void derefGPtr<GstPlugin>(GstPlugin* ptr);
+#endif  // USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 
 template<> GRefPtr<GstMiniObject> adoptGRef(GstMiniObject* ptr);
 template<> GstMiniObject* refGPtr<GstMiniObject>(GstMiniObject* ptr);
@@ -162,9 +162,11 @@ template<> GRefPtr<GstGLContext> adoptGRef(GstGLContext* ptr);
 template<> GstGLContext* refGPtr<GstGLContext>(GstGLContext* ptr);
 template<> void derefGPtr<GstGLContext>(GstGLContext* ptr);
 
+#if USE(WPE_VIDEO_PLANE_DISPLAY_DMABUF)
 template<> GRefPtr<GstEGLImage> adoptGRef(GstEGLImage* ptr);
 template<> GstEGLImage* refGPtr<GstEGLImage>(GstEGLImage* ptr);
 template<> void derefGPtr<GstEGLImage>(GstEGLImage* ptr);
+#endif
 
 template<> GRefPtr<GstGLColorConvert> adoptGRef(GstGLColorConvert* ptr);
 template<> GstGLColorConvert* refGPtr<GstGLColorConvert>(GstGLColorConvert* ptr);
